### PR TITLE
rpc wallet

### PR DESCRIPTION
**Pull request by yz89**

### CODE REVIEW

File: wallet/filestore.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+// Caller holds the lock and reads bytes from DB, then close the DB and release the lock
+func (cs *FileStore) readDB() ([]byte, error) {
+	cs.Lock()
+	defer cs.Unlock()
+	defer cs.closeDB()
+
+	var err error
+	cs.file, err = os.OpenFile(cs.path, os.O_RDONLY, 0666)
```
Comment (raindust) [inline wallet/filestore.go]: file permission set to 0400 would be better

File: wallet/filestore.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+// Caller holds the lock and writes bytes to DB, then close the DB and release the lock
+func (cs *FileStore) writeDB(data []byte) error {
+	cs.Lock()
+	defer cs.Unlock()
+	defer cs.closeDB()
+
+	var err error
+	cs.file, err = os.OpenFile(cs.path, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0666)
```
Comment (raindust) [inline wallet/filestore.go]: file permission set to 0600 would be better

File: wallet/wallet.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
+package wallet
+
+import (
+	"errors"
+	"github.com/elastos/Elastos.ELA/common"
```
Comment (raindust) [inline wallet/wallet.go]: format imports

File: wallet/filestore.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+}
+
+// Caller holds the lock and writes bytes to DB, then close the DB and release the lock
+func (cs *FileStore) writeDB(data []byte) error {
+	cs.Lock()
+	defer cs.Unlock()
+	defer cs.closeDB()
+
+	var err error
+	cs.file, err = os.OpenFile(cs.path, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0640)
```
Comment (peterwillcn) [inline wallet/filestore.go]: try 0740

**Pull request merged**